### DISCUSSION TITLE
Development workflow when working on multiples app/package at the same time

### DISCUSSION THREAD

**Discussion by AnatoleLucet** [vercel/turborepo#772] [Ideas]
Hi 👋 I'd like to ask a few questions about how this project could be used when working on multiples apps/packages at the same time.

TL;DR: does it make sens for turborepo to start related apps (like starting the api when running the frontend in dev mode), and either  start the third-party apps (the api)  in preview/prod mode; or in dev mode if the app (api) is started manually?

---

Let's say I have the following monorepo:
```
apps/
  api/ <- depends on utils
  web-app/ <- depends on svelte-ui & web-services & utils
  mobile-app/ <- depends on react-native-ui & web-services & utils
packages/
  svelte-ui/
  react-native-ui/
  web-services/ <- a collection of internal & thrid-party api calls
  utils/
```

Is there a way to tell turborepo that, if I start the `web-app`, it needs to do the following? :
- start the api
  - build all its dependencies
- start the web-app
  - build all its dependencies

Also, what if I want to make a few changes on `svelte-ui`, would I need to restart the turborepo process? Or is there a way to build the dependencies in dev/watch mode?

Something I'm also not sure is: should the api be run in dev mode, or built and previewed to mimic production?
I think:
- if the `web-app` is started in dev: build and preview the api (mimic prod)
- if the `api` is started in dev beside the `web-app`: stop the "build and preview" api, and run a new api in dev mode

Does this make any sens, or am I overthinking this?

**Comment by ghost** [comment]
Any update on these. I also need to update the server when a common dependant get updated.

**Comment by alexn-s** [comment]
i am also interested in this topic

**Comment by anthonyshew** [comment]
Either approach is okay, depending on the workflow you'd like to create. Both dev servers can be spinning at the same time and you would need a watcher development server for your apps.

If you'd like Turborepo to be a helper for your development servers, make sure to upvote this issue (if you haven't already): https://github.com/vercel/turbo/issues/986
